### PR TITLE
Wallet completions, CLI extensions, UTESTs created

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/plugins/witness/witness_plugin.cpp [hunk 1]
```diff
@@ -53,15 +55,41 @@ using chain::account_object;
...
-                 "\n";
+    const std::string welcome = "Welcome to Scorum!";
+    const std::string info_ch = "NEW CHAIN";
+    const char space_symbol = ' ';
+    const char border_symbol = '*';
+
+    size_t sz = std::max(welcome.size(), info_ch.size());
+    sz *= 3;
+    cli::formatter p(sz);
+    p.print_line(border_symbol);
```
Comment (alexplevako) [inline libraries/plugins/witness/witness_plugin.cpp]: this implementation looks heavy, over complicated for me.

CliBuffer().println("NEW CHAIN")
                .println("Welcom to Scorum")
                .frame("*")
                .margin(1);

File: programs/cli_wallet/wallet_app.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    completions_incontext_type completions = { "0." + completion_mask + " SCR", "0." + completion_mask + " SP" };
+    std::string completion_reg(R"(\d+\.)");
+    for (int ci = 0; ci < SCORUM_CURRENCY_PRECISION + 1; ++ci)
+    {
+        registry_completions(completions, cli::completion_context(completion_reg, true));
+        completion_reg.append(R"(\d)");
+    }
+    registry_completions("SCR\"", cli::completion_context(R"(.+"\d+\.\d+\s+SC)", R"(SC)", false));
+    registry_completions("SP\"", cli::completion_context(R"(.+"\d+\.\d+\s+SP)", R"(SP)", false));
+}
```
Comment (alexplevako) [inline programs/cli_wallet/wallet_app.cpp]: Let's move this code in register_asset_completion()

File: programs/cli_wallet/wallet_app.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+void wallet_app::start()
+{
+    completions_incontext_type commands = get_method_names(0);
```
Comment (alexplevako) [inline programs/cli_wallet/wallet_app.cpp]: lets create function register_remote_api_completion

File: tests/common/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    message(STATUS "Found gperftools; compiling tests with TCMalloc")
+    list(APPEND PLATFORM_SPECIFIC_LIBS tcmalloc)
+endif()
+
+set( SOURCES
+  actor.cpp
+  database_fixture.cpp
+)
+
+set( HEADERS
```
Comment (abolinbsu) [inline tests/common/CMakeLists.txt]: Use GLOB for *.h files

**Pull request merged**